api/src/main/java/org/hawkular/apm/api/services/Criteria.java

```diff
@@ -139,9 +140,9 @@ public Criteria setEndTime(long endTime) {
      */
     public long calculateEndTime() {
         if (endTime == 0) {
-            return System.currentTimeMillis();
+            return TimeUnit.NANOSECONDS.toMicros(System.nanoTime());
```

Comment:
Brave uses this https://github.com/pavolloffay/brave/blob/f511bc645590c116fca8170a69def6c2ef4f4e58/brave-core/src/main/java/com/github/kristofa/brave/AnnotationSubmitter.java#L150-L150 however it allows to submit annotations with timestamp precision higher than milliseconds. It seems that in JVM the safest way is to use `currentTimeMillis()`, we will loose some accuracy in our agent, but other clients (zipkin/opentracing) could use better precision.